velox/exec/tests/SpillTest.cpp

```diff
@@ -42,46 +43,155 @@ class SpillTest : public testing::Test,
 
 TEST_F(SpillTest, spillState) {
   auto tempDirectory = exec::test::TempDirectoryPath::create();
+  const std::string kSpillPath = tempDirectory->path + "/test";
+  const int kNumPartitions = 2;
+  // Set the target file size to a large value to avoid new file creation on
+  // each batch write.
+  const int kTargetFileSize = 1'000'000'000;
+  std::shared_ptr<FileSystem> fs;
+  std::unordered_set<std::string> spilledFileSet;
+  {
+    // We make a state that has 2 partitions, each with its own file
+    // list. We write 10 sorted vectors in each partition. The vectors
+    // have the ith element = i * 20 + sequence, where sequence is the
+    // sequence number of the vector in the partition. When read back,
+    // both partitions produce an ascending sequence of integers without
+    // gaps.
+    SpillState state(
+        kSpillPath,
+        kNumPartitions,
+        1,
+        kTargetFileSize,
+        *pool(),
+        *mappedMemory_);
+    EXPECT_EQ(kTargetFileSize, state.targetFileSize());
+    EXPECT_EQ(kNumPartitions, state.maxPartitions());
+
+    int expectedSpilledFiles = 0;
+    for (auto partition = 0; partition < state.maxPartitions(); ++partition) {
+      EXPECT_FALSE(state.isPartitionSpilled(partition));
+#ifndef NEDBUG
```

Comment:
Why do we need this?

Reply:
I try to trigger a VELOX_DCHECK in production code so wrap around with NEDBUG. I change it to CHECK since it is not hot path. It might be better to fail the process instead of letting it proceed in this case.